doc/test-framework/writing-e2e-tests.md

```diff
@@ -0,0 +1,206 @@
+file simply calls the test framework's main entry that sets up the framework and then
+starts the tests. It should be pretty much identical for all operators. This is what it
+looks like for the memcached-operator:
+```
+package e2e
+
+import (
+    "testing"
+
+    f "github.com/operator-framework/operator-sdk/pkg/test"
+)
+
+func TestMain(m *testing.M) {
+    f.MainEntry(m)
+}
+```
+
+### Individual Tests
+Once MainEntry sets up the framework, it runs the remainder of the tests. First, make
+sure to import `testing`, the operator-sdk test framework (`pkg/test`) as well as your operator's libraries:
+```
+import (
+    "testing"
+
+    cachev1alpha1 "github.com/operator-framework/operator-sdk/test/test-framework/pkg/apis/cache/v1alpha1"
+
+    framework "github.com/operator-framework/operator-sdk/pkg/test"
+)
+```
+
+The next step is to register your operator's scheme with the framework's dynamic client.
+To do this, you need to create a list struct for your custom resource and pass the list
+object and the object's addToScheme function to the framework's AddToFrameworkScheme
+function. For our example memcached-operator, it looks like this:
+```
+memcachedList := &cachev1alpha1.MemcachedList{
+    TypeMeta: metav1.TypeMeta{
+        Kind:       "Memcached",
+        APIVersion: "cache.example.com/v1alpha1",
+    },
+}
+err := framework.AddToFrameworkScheme(cachev1alpha1.AddToScheme, memcachedList)
+if err != nil {
+    t.Fatalf("failed to add custom resource scheme to framework: %v", err)
+}
+```
+
+The reason that the list struct is necessary for this function is because the dynamic client's RESTMapper
+needs to have the custom resource's mappings to work. These mappings are created by the kubernetes
+cluster. Since the CRD is created as the last step of the framework setup process, it is likely that the
+resource will not be ready by the time the first test starts. By providing the custom resource's list
+struct, the AddToFrameworkScheme function can verify that the dynamic client has the REST mappings for
+your custom resource. If the mappings are not yet available, the framework simply waits and attempts again
+1 second later. If the mappings are still not available after 5 seconds, the framework assumes something
+went wrong and returns an error.
+
+The next step is to create a TestCtx for the current test and defer its cleanup function:
+```
+ctx := framework.NewTestCtx(t)
+defer ctx.Cleanup(t)
+```
+
+Now that there is a TestCtx, the test's kubernetes resources (specifically the RBAC and Operator deployment)
+can be initialized:
+```
+err := ctx.InitializeClusterResources()
+if err != nil {
+    t.Fatalf("failed to initialize cluster resources: %v", err)
+}
+```
+
+If you want to make sure the operator's deployment is fully ready before moving onto the next part of the
+test, the `WaitForDeployment` function from e2eutil can be used. First, import `"github.com/operator-framework/operator-sdk/pkg/util/e2eutil"`
```

Comment:
Let's move `github.com/operator-framework/operator-sdk/pkg/util/e2eutil` to `github.com/operator-framework/operator-sdk/pkg/test/e2eutil`
Since its not a general util used by the SDK itself. And anything that's meant to be used as part of the test-framework should be under `github.com/operator-framework/operator-sdk/pkg/test`